src/components/AnonymousChat/child/ChatItemMyTextV2.tsx

```diff
@@ -144,6 +154,81 @@ const ChatItemMyTextV2 = ({
+                                <Text style={styles.attachmentFileName}>{item.file_name}</Text>
+                                <View>
+                                  <Text style={styles.attachmentFileInfo}>
+                                    {formatBytes(item.file_size)} •{' '}
+                                    {item.file_name
+                                      ?.split('.')
+                                      [item.file_name?.split('.')?.length - 1]?.toUpperCase()}
+                                  </Text>
+                                </View>
+                              </View>
+                              <View style={styles.attachmentFileIcon}>
+                                <IconFile />
+                              </View>
+                            </View>
+                          </TouchableOpacity>
+                        ) : (
+                          <TouchableOpacity
+                            key={item.thumb_url}
+                            style={{
+                              width: `${
+                                (attachments.length >= 3 && index > 0) || attachments.length >= 4
+                                  ? 50
+                                  : 100
+                              }%`,
+                              height: `${attachments.length >= 3 ? 50 : 100 / attachments.length}%`,
+                              position: 'relative',
+                              overflow: 'hidden'
+                            }}
+                            activeOpacity={1}
+                            onPress={
+                              attachments.length > 0 && item.type !== 'gif'
+                                ? () => onOpenMediaPreview(attachments, index, navigation)
+                                : null
+                            }>
+                            {item.type !== 'file' && (
+                              <FastImage
+                                style={styles.image}
+                                source={{
+                                  uri: item.thumb_url
+                                }}
+                              />
+                            )}
+                            {attachments.length > 4 && index === 3 && (
+                              <View style={styles.moreOverlay}>
+                                <Text style={styles.moreText}>+{attachments.length - 4}</Text>
+                              </View>
+                            )}
+                            {/* Video Play Icon */}
+                            {item.video_path && (
+                              <View style={styles.moreOverlay}>
+                                <IconVideoPlay />
+                              </View>
+                            )}
+                          </TouchableOpacity>
+                        )
+                      )}
+                  </View>
+                )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block handles the rendering of attachments. It's quite large and complex, which could make it difficult to maintain. Consider breaking it down into smaller, more manageable functions or even separate components. For example, you could create separate components for file and image previews.

```diff
- {attachments
-   .filter((item, index) => index <= 3)
-   .map((item, index) =>
-     item.type === 'file' ? (
-       <TouchableOpacity
-         key={item.file_path}
-         style={{flex: 1}}
-         activeOpacity={1}
-         onPress={() => Linking.openURL(item.file_path)}>
-         <View style={styles.attachmentFileContainer}>
-           <View style={styles.attachmentFileContent}>
-             <Text style={styles.attachmentFileName}>{item.file_name}</Text>
-             <View>
-               <Text style={styles.attachmentFileInfo}>
-                 {formatBytes(item.file_size)} •{' '}
-                 {item.file_name
-                   ?.split('.')
-                   [item.file_name?.split('.')?.length - 1]?.toUpperCase()}
-               </Text>
-             </View>
-           </View>
-           <View style={styles.attachmentFileIcon}>
-             <IconFile />
-           </View>
-         </View>
-       </TouchableOpacity>
-     ) : (
-       <TouchableOpacity
-         key={item.thumb_url}
-         style={{
-           width: `${
-             (attachments.length >= 3 && index > 0) || attachments.length >= 4
-               ? 50
-               : 100
-           }%`,
-           height: `${attachments.length >= 3 ? 50 : 100 / attachments.length}%`,
-           position: 'relative',
-           overflow: 'hidden'
-         }}
-         activeOpacity={1}
-         onPress={
-           attachments.length > 0 && item.type !== 'gif'
-             ? () => onOpenMediaPreview(attachments, index, navigation)
-             : null
-         }>
-         {item.type !== 'file' && (
-           <FastImage
-             style={styles.image}
-             source={{
-               uri: item.thumb_url
-             }}
-           />
-         )}
-         {attachments.length > 4 && index === 3 && (
-           <View style={styles.moreOverlay}>
-             <Text style={styles.moreText}>+{attachments.length - 4}</Text>
-           </View>
-         )}
-         {/* Video Play Icon */}
-         {item.video_path && (
-           <View style={styles.moreOverlay}>
-             <IconVideoPlay />
-           </View>
-         )}
-       </TouchableOpacity>
-     )
-   )}
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->